cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -703,6 +707,25 @@ TEST_P(TestParquetFileFormatScan, PredicatePushdownRowGroupFragmentsUsingStringC
   CountRowGroupsInFragment(fragment, {0, 3}, equal(field_ref("x"), literal("a")));
 }
 
+TEST_P(TestParquetFileFormatScan, PredicatePushdownRowGroupFragmentsUsingDurationColumn) {
+  auto table = TableFromJSON(schema({field("t", duration(TimeUnit::NANO))}),
```

Comment:
Would you mind also checking for time32 since that was mentioned in the issue? If it isn't fixed by this patch then we should make a follow up issue to address it

```suggestion
  for (auto type : {duration(TimeUnit::NANO), time32(TimeUnit::SECOND)}) {
    auto table = TableFromJSON(schema({field("t", type)}),
```

Also, please describe the bug in GH-37111 and how this test checks it

Reply:
I think this is a different problem, after check this patch not fixed that, I'll create another issue and try to fix it before 14.0